api/v1alpha1/hostedcluster_webhook.go

```diff
@@ -0,0 +1,49 @@
+package v1alpha1
+
+import (
```

Comment:
The generic issue with this approach to couple more deps with api/ package.

Reply:
Is there a way to avoid this?

Reply:
Something like this: https://github.com/kubernetes-sigs/controller-runtime/blob/master/examples/builtins/validatingwebhook.go